### PR TITLE
GRE-106 add color processing

### PR DESCRIPTION
**Pull request by sknowlton-everi**

### CODE REVIEW

**Review by roatanrich — CHANGES_REQUESTED** [review]
make sure to add requested comments and test categories

File: GliToJiraImporter.Testing/Tests/ParserTester.cs [hunk 1]
```diff
@@ -209,5 +209,36 @@ public void ParserLinkTest()
...
+            IList<CategoryModel> expectedResult = JsonSerializer.Deserialize<List<CategoryModel>>(File.ReadAllText(expectedResultPath));
+
+            //when
+            IList<CategoryModel> result = this.sut.Parse();
+
+            //then
+            expectedResult.TestAssertCategoryModels(result);
+        }
+
+        [Ignore("This test is very large and takes a long time to complete.")]
```
[the remark this answers is not in the document]
Comment (sknowlton-everi) [reply GliToJiraImporter.Testing/Tests/ParserTester.cs]: That message should have been changed has it isn't accurate. It has been fixed

File: GliToJiraImporter/Parsers/Parser.cs [hunk 2]
```diff
@@ -60,45 +65,49 @@ private IList<CategoryModel> parseMementos()
...
-                        }
-                        else if (categoryOriginator.Save().IsValid())
-                        {
-                            log.Debug("Category Parsing valid");
-                            result.Add((CategoryModel)categoryOriginator.Save());
-                            categoryOriginator = new CategoryParser();
-                        }
+                    if (!isCategoryComplete) continue;
+
+                    j--;
```
[the remark this answers is not in the document]
Comment (sknowlton-everi) [reply GliToJiraImporter/Parsers/Parser.cs:100]: Good call, I added comments to all the ones I could remember

File: GliToJiraImporter/Parsers/RegulationParser.cs [hunk 3]
```diff
@@ -39,38 +45,85 @@ public bool Parse(WTableRow row)
...
+                                regulationModel.ClauseIdModels[clauseIdParser.DocumentColor.ToArgb()] = (ClauseIdModel)clauseIdParser.Save();
+                            }
+                            else
+                            {
+                                clauseIdColors.Add(clauseIdParser.DocumentColor);
+                                regulationModel.ClauseIdModels.Add(clauseIdParser.DocumentColor.ToArgb(), (ClauseIdModel)clauseIdParser.Save());
+                            }
+                            if (!parsedFully)
+                            {
+                                i--;
```
[the remark this answers is not in the document]
Comment (sknowlton-everi) [reply GliToJiraImporter/Parsers/RegulationParser.cs:79]: Done

File: GliToJiraImporter/Parsers/Parser.cs [hunk 4]
```diff
@@ -128,7 +137,67 @@ private WordDocument getDocumentFromPath()
...
-        private int getMainTableIndex(EntityCollection documentBodyEntities)
+        private IDictionary<string, Color> parseDocumentColors(IWTableCollection tables)
+        {
+            IDictionary<string, Color> result = new Dictionary<string, Color>();
+
+            IWTable? table = null;
+            // Iterates through the entities of the section to find the "Tested against Requirements" table
+            for (int i = 0; i < tables.Count; i++)
+            {
+                WTableCell firstCell = tables[i].Rows[0].Cells[0];
```
[the remark this answers is not in the document]
Comment (sknowlton-everi) [reply GliToJiraImporter/Parsers/Parser.cs:151]: I was hoping the name would be enough as you have mentioned a preference for that before. Comment added, let me know if it's okay

**Pull request merged**